bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsBinarySyncImpl.java

```diff
@@ -0,0 +1,201 @@
+    protected void deactivate() {
+        if (this.session != null) {
+            try {
+                this.session.logout();
+            } catch (Exception e) {
+                LOG.warn("Failed session.logout()", e);
+            }
+        }
+        if (this.resourceResolver != null) {
+            try {
+                this.resourceResolver.close();
+            } catch (Exception e) {
+                LOG.warn("Failed resourceResolver.close()", e);
+            }
+        }
+    }
+
+    /**
+     * @see RemoteAssetsBinarySync#syncAsset(Resource)
+     * @param resource Resource
+     * @return Resource
+     */
+    @Override
+    public Resource syncAsset(Resource resource) {
+        try {
+            this.session.refresh(true);
+            Resource localRes = this.resourceResolver.getResource(resource.getPath());
+            Node node = localRes.adaptTo(Node.class);
+
+            Asset asset = localRes.getParent().adaptTo(Asset.class);
+            URI pathUi = new URI(null, null, asset.getPath(), null);
+            String baseUrl = this.remoteAssetsConfig.getServer().concat(pathUi.toString()).concat("/_jcr_content/renditions/");
+
+            Iterator<? extends Rendition> renditions = asset.listRenditions();
+            while (renditions.hasNext()) {
+                Rendition assetRendition = renditions.next();
+                if (StringUtils.isEmpty(assetRendition.getMimeType())) {
+                    continue;
+                }
+                String renditionName = assetRendition.getName();
+
+                URL url = new URL(String.format("%s%s", baseUrl, renditionName));
+                HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+                connection.setRequestMethod("GET");
+                String rawAuth = String.format("%s:%s", this.remoteAssetsConfig.getUsername(), this.remoteAssetsConfig.getPassword());
+                String encodedAuth = Base64.getEncoder().encodeToString(rawAuth.getBytes(StandardCharsets.UTF_8));
+                connection.setRequestProperty("Authorization", String.format("Basic %s", encodedAuth));
+
+                LOG.info("syncing from remote asset url {}", url);
+
+                InputStream inputStream;
+                try {
+                    inputStream = connection.getInputStream();
+                } catch (FileNotFoundException fne) {
+                    if ("original".equals(renditionName)) {
+                        throw fne;
+                    }
+
+                    asset.removeRendition(renditionName);
+                    LOG.warn("Rendition '{}' not found on remote environment. Removing local rendition.", renditionName);
+                    continue;
+                }
+
+                Map<String, Object> props = new HashMap<>();
+                props.put("rendition.mime", assetRendition.getMimeType());
+                asset.setRendition(renditionName, inputStream, props);
+
+                inputStream.close();
```

Comment:
Should probably close this in a `finally` block.  Also, do we need to close the `connection`?